use/mobile-gaming-example.md

```diff
@@ -0,0 +1,540 @@
+            -> gInfo.getTimestamp() > startMinTimestamp.getMillis()))
+    .apply("FilterEndTime", Filter.by(
+        (GameActionInfo gInfo)
+            -> gInfo.getTimestamp() < stopMinTimestamp.getMillis()))
+    // [END DocInclude_HTSFilters]
+
+    // [START DocInclude_HTSAddTsAndWindow]
+    // Add an element timestamp based on the event log, and apply fixed windowing.
+    .apply("AddEventTimestamps",
+           WithTimestamps.of((GameActionInfo i) -> new Instant(i.getTimestamp())))
+    .apply("FixedWindowsTeam", Window.<GameActionInfo>into(
+        FixedWindows.of(Duration.standardMinutes(options.getWindowDuration()))))
+    // [END DocInclude_HTSAddTsAndWindow]
+
+    // Extract and sum teamname/score pairs from the event data.
+    .apply("ExtractTeamScore", new ExtractAndSumScore("team"))
+    .apply("WriteTeamScoreSums",
+      new WriteWindowedToBigQuery<KV<String, Integer>>(options.getTableName(),
+          configureWindowedTableWrite()));
+
+  pipeline.run().waitUntilFinish();
+}
+```
+
+### Limitations
+
+As written, `HourlyTeamScore` still has a limitation:
+
+* `HourlyTeamScore` still has **high latency** between when data events occur (the event time) and when results are generated (the processing time), because, as a batch pipeline, it needs to wait to begin processing until all data events are present.
+
+
+## LeaderBoard: Streaming Processing with Real-Time Game Data
+
+One way we can help address the latency issue present in the `UserScore` and `HourlyTeamScore` pipelines is by reading the score data from a streaming source. The `LeaderBoard` pipeline introduces streaming processing by reading the game score data from [Google Cloud Pub/Sub](https://cloud.google.com/pubsub/), rather than from a file on the game server.
+
+> **Note:** For the `LeaderBoard` pipeline to use Pub/Sub as an input source, the game server would need to actively publish the score data to Pub/Sub as it arrives, rather than simply logging it and writing it to a file.
+
+The `LeaderBoard` pipeline also demonstrates how to process game score data with respect to both _processing time_ and _event time_. `LeaderBoard` outputs data about both individual user scores and about team scores, each with respect to a different time frame.
+
+Because the `LeaderBoard` pipeline reads the game data from a streaming source as that data is generated, you can think of the pipeline as an ongoing job running concurrently with the game process. `LeaderBoard` can thus provide low-latency insights into how users are playing the game at any given moment—useful if, for example, we want to provide a live web-based scoreboard so that users can track their progress against other users as they play.
+
+> **Note:** See [LeaderBoard on GitHub](https://github.com/apache/incubator-beam/blob/ee6ad2fe416cd499aa1b6e2a51aa64da0805cc5c/examples/java8/src/main/java/org/apache/beam/examples/complete/game/LeaderBoard.java) for the complete example pipeline program.
```

Comment:
all of these links are linking to a specific revision of these files. Should be changed to a general link.